### PR TITLE
BUGFIX: SD-901 fix switch to geo chart with derived and arithmetic measures

### PR DESCRIPTION
**Pull request by lanhq**
---

[Check PR owner responsibilities](https://confluence.intgdc.com/display/Development/Code-reviews#Code-reviews-Ownerresponsibilities)

Supported PR commands:

Command | Description
--- | ---
`extended test - examples` | Live examples tests
`extended test - storybook` | Storybook screenshot tests

See more [options](https://confluence.intgdc.com/display/kbhr/How+to+work+with+git+and+Github#HowtoworkwithgitandGithub-Extendedchecks).

# PR Checklist

- [x] Verify code changes ([Checklist](https://confluence.intgdc.com/display/Development/Code-reviews+checklist), [Best practices](https://confluence.intgdc.com/display/Development/Code-reviews+best+practices))
- [x] [Verify pull-request formalities](https://confluence.intgdc.com/display/Development/Code-reviews)
- [x] Change was tested by using [gdc-dev-release](https://confluence.intgdc.com/display/~tomas.vojtasek/Private+NPM) in [gdc-analytical-designer](https://github.com/gooddata/gdc-analytical-designer) and [gdc-dashboards](https://github.com/gooddata/gdc-dashboards) (if applicable)
- [x] Migration guide (for major changes) is mentioned in [CHANGELOG.md](../blob/master/CHANGELOG.md).
- [x] Successful `extended test - examples`
- [ ] Successful `extended test - storybook`
- [x] Checked yarn.lock consistency (no dep. duplicities especially Goodstrap)

# Related PRs

- gdc-analytical-designer: https://github.com/gooddata/gdc-analytical-designer/pull/2991
- gdc-dashboards: https://github.com/gooddata/gdc-dashboards/pull/2706

# Related Jira tasks
- SD-901: https://jira.intgdc.com/browse/SD-901

### CODE REVIEW

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by lanhq** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6901/ : SUCCESS in 9m 43s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2788/ : SUCCESS in 3s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3600/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2789/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2970/ : UNSTABLE in 18m 31s

**Comment by lanhq** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1519.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by ptquang86** [comment]
LGTM

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2793/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2972/ : SUCCESS in 24m 50s

**Comment by lanhq** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3603/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2796/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2975/ : UNSTABLE in 21m 46s

File: src/internal/components/pluggableVisualizations/geoChart/PluggableGeoPushpinChart.tsx [hunk 1]
```diff
@@ -78,7 +83,14 @@ export class PluggableGeoPushpinChart extends PluggableBaseChart {
     }
 
     protected configureBuckets(extendedReferencePoint: IExtendedReferencePoint): IExtendedReferencePoint {
-        const buckets: IBucket[] = get(extendedReferencePoint, BUCKETS, []);
+        let newExtendedReferencePoint: IExtendedReferencePoint = removeAllArithmeticMeasuresFromDerived(
+            extendedReferencePoint,
+        );
+        newExtendedReferencePoint = removeAllDerivedMeasures(newExtendedReferencePoint);
```
Comment (naibinh) [inline src/internal/components/pluggableVisualizations/geoChart/PluggableGeoPushpinChart.tsx:89]: We should extract line 86->89 to `sanitizeMeasure` method

```
private sanitizeMeasures(extendedReferencePoint: IExtendedReferencePoint) : IExtendedReferencePoint {...}
```

**Comment by lanhq** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6905/ : SUCCESS in 4m 59s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2799/ : SUCCESS in 3s

**Comment by naibinh** [comment]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check-extended aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3607/ : SUCCESS in 15m 28s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6910/ : SUCCESS in 3m 14s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/1675/ : SUCCESS in 14s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/2424/ : SUCCESS in 17m 20s

**Pull request merged**